src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs

```diff
@@ -84,16 +86,25 @@ public IEnumerable<LibraryExport> GetDependencies(LibraryType type)
         /// </summary>
         private IEnumerable<LibraryExport> ExportLibraries(Func<LibraryDescription, bool> condition)
         {
-            var seenMetadataReferences = new HashSet<string>();
+            IEnumerable<LibraryExport> exports = _cachedExports ?? (_cachedExports = CalculateAllExports().ToArray());
```

Comment:
Why not:

_cachedExports = _cachedExports ?? CalculateAllExports().ToArray();

And then just use the field in the foreach below.